Update enterprise advantages

nickoferrall opened the pull request:
Fix https://github.com/ParabolInc/parabol/issues/8572

![Screenshot 2023-07-27 at 14 34 08](https://github.com/ParabolInc/parabol/assets/39854876/227a762c-0910-4661-9c02-bbde8ad73971)

### To test

- [ ] Add the checkout flow feature flag
- [ ] See the org plans view has the updated copy

acressall commented:
While we're at it, should we add "health checks" to the team description? Maybe swap out "priority customer support"?

nickoferrall commented:
Yeah, good point @acressall. I added Team Health and added Jira Server back to Enterprise

![Screenshot 2023-07-27 at 16 23 21](https://github.com/ParabolInc/parabol/assets/39854876/2bab3209-90b2-46de-b434-a920733d70c1)
